internal/operands/tekton-cleanup/reconcile.go

```diff
@@ -0,0 +1,175 @@
+package tekton_cleanup
+
+import (
+	"fmt"
+
+	libhandler "github.com/operator-framework/operator-lib/handler"
+	pipeline "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands"
+)
+
+// +kubebuilder:rbac:groups=tekton.dev,resources=pipelines,verbs=list;watch;create;update
+// +kubebuilder:rbac:groups=tekton.dev,resources=tasks,verbs=list;watch;update
+// +kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=list;watch;update
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;rolebindings,verbs=list;watch;update
+// +kubebuilder:rbac:groups=core,resources=configmaps,verbs=list;watch;create;update
+
+// Below are RBAC for deployed ClusterRoles. We still need these permissions so we can update annotations on existing ClusterRoles
+
+// RBAC rules for "cleanup-vm-task"
+// +kubebuilder:rbac:groups=kubevirt.io,resources=virtualmachineinstances;virtualmachines,verbs=get;list;watch;delete
+// +kubebuilder:rbac:groups=subresources.kubevirt.io,resources=virtualmachines/restart;virtualmachines/start;virtualmachines/stop,verbs=update
+
+// RBAC rules for "copy-template-task"
+// +kubebuilder:rbac:groups=template.openshift.io,resources=templates,verbs=get;list;watch;create;update
```

Comment:
Are all of these rbacs necessary? Since operator will not deploy anything and just udate each object.

Reply:
Yes, I've tried it. When the operator updates a `ClusterRole` it needs to have all the RBAC permissions that are in the `ClusterRole`.

Reply:
I will try to use `Patch` to only change the annotations.

Reply:
I've tried using `Patch` instead of `Update` but the RBAC permissions are still required.

Reply:
I've tried using `Patch` instead of `Update` but the RBAC permissions are still required.